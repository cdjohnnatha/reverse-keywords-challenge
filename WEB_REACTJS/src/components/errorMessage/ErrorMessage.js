import React from 'react';
import { number, func } from 'prop-types';

const styles = {
  spanStyle: {
    color: 'red',
  },
  errorMessageContainer: {
    marginTop: '1em',
  },
};

const ErrorMessage = ({
  message,
  hideMessageTimeout,
  cleanErrorMessageHandler,
}) => {
  const { spanStyle, errorMessageContainer } = styles;

  let renderMessageComponent = null;
  if (message) {
    renderMessageComponent = (
      <div style={errorMessageContainer}>
        <span style={spanStyle}>An error happened, {message}</span>
      </div>
    );
    setTimeout(() => {
      cleanErrorMessageHandler(undefined);
    }, hideMessageTimeout);
  }

  return renderMessageComponent;
};

ErrorMessage.propTypes = {
  hideMessageTimeout: number,
  cleanErrorMessageHandler: func.isRequired,
};

ErrorMessage.defaultProps = {
  hideMessageTimeout: 2000,
};

export default ErrorMessage;
