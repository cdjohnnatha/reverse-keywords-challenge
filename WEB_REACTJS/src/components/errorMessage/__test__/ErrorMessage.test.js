import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import ErrorMessage from '../ErrorMessage';

import '../../../../jest.setup';

describe('Test ErrorMessage component', () => {
  test('ErrorMessage snapshot renders', () => {
    const wrapper = create(
      <ErrorMessage
        message="Network error"
        cleanErrorMessageHandler={function () {}}
      />
    );
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ErrorMessage should show a message when its set', () => {
    const wrapper = mount(
      <ErrorMessage
        message="Network error"
        cleanErrorMessageHandler={function () {}}
      />
    );
    expect(wrapper.props().message).toBe('Network error');
    // expect(cardsContainer.getElement().type).toBe('div');
  });
});
