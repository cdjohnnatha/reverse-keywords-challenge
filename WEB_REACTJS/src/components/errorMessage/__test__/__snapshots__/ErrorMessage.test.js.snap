// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ErrorMessage component ErrorMessage snapshot renders 1`] = `
<div
  style={
    Object {
      "marginTop": "1em",
    }
  }
>
  <span
    style={
      Object {
        "color": "red",
      }
    }
  >
    An error happened, 
    Network error
  </span>
</div>
`;
